pkg/bootstrapper/bootstrapper.go

```diff
@@ -211,51 +203,44 @@ type fileTranslation struct {
-	b := bufio.NewReader(bytes.NewReader(initialConfig))
-	r := yaml.NewYAMLReader(b)
-	doc, err := r.Read()
+// kubeletConf defines fileds of kubelet.conf file that are defined by wmcb variables
+type kubeletConf struct {
+	// ClientCAFile specifies location to client certificate
+	ClientCAFile string
+}
+
+// prepKubeletConfForWindows creates config file for kubelet, with windows specific configuration
+// Add values in kubelet_config.json files, for additional static fields.
+// Add fields in kubeletConf struct for variable fields
+func prepKubeletConfForWindows(wmcb *winNodeBootstrapper) ([]byte, error) {
+	// get config file content using bindata.go
+	content, err := Asset("kubelet_config.json")
+
 	if err != nil {
-		return out, err
+		return nil, fmt.Errorf("error reading kubelet config template: %v", err)
 	}
-	scheme := runtime.NewScheme()
-	err = kubeletConfig.AddToScheme(scheme)
+	kubeletConfTmpl := template.New("kubeletconf")
+
+	// Parse the template
+	kubeletConfTmpl, err = kubeletConfTmpl.Parse(string(content))
 	if err != nil {
-		return out, err
+		return nil, err
+	}
+	// Fill up the config file, using kubeletConf struct
+	variableFields := kubeletConf{
+		ClientCAFile: strings.Join(append(strings.Split(wmcb.installDir, `\`), `kubelet-ca.crt`), `\\`),
 	}
-	d := serializer.NewCodecFactory(scheme).UniversalDeserializer()
-	config := kubeletConfig.KubeletConfiguration{}
-	_, _, err = d.Decode(doc, nil, &config)
+	// Create kubelet.conf file
+	kubeletConfigPath, err := os.Create(filepath.Join(wmcb.installDir, "kubelet.conf"))
 	if err != nil {
-		return out, fmt.Errorf("could not decode yaml: %s\n%s", initialConfig, err)
+		return nil, fmt.Errorf("error creating kubelet.conf: %v", err)
 	}
-
-	// Here we edit the config's fields so we can run the kubelet on Windows
-	config.CgroupDriver = "cgroupfs"
-	config.ResolverConfig = ""
-	cgroupsPerQOS := false
-	config.CgroupsPerQOS = &cgroupsPerQOS
-	config.Authentication.X509.ClientCAFile = filepath.Join(wmcb.installDir, "kubelet-ca.crt")
-
-	// We need to set EnforceNodeAllocatable with an empty slice, "enforceNodeAllocatable:[]"
-	// the json tags have the field set as `omitempty`, and the field defaults to enforceNodeAllocatable:["pods"]
-	// so we need empty the slice after marshalling
-	// Putting a placeholder value here to be removed later in the function
-	config.EnforceNodeAllocatable = []string{"THIS_MUST_BE_EMPTY"}
-
-	// Turn the config into a json marshalled []byte
-	out, err = json.Marshal(config)
+	err = kubeletConfTmpl.Execute(kubeletConfigPath, variableFields)
 	if err != nil {
-		return out, err
+		return nil, fmt.Errorf("error writing data to kubelet.conf file: %v", err)
 	}
-
-	// replacing EnforceNodeAllocatable with an empty slice,
-	outString := strings.Replace(string(out), "\"THIS_MUST_BE_EMPTY\"", "", -1)
-	return []byte(outString), err
+	data, _ := ioutil.ReadFile(kubeletConfigPath.Name())
```

Comment:
please add an error check for the read , data can be renamed to something specific